core/src/main/java/org/elasticsearch/indices/query/IndicesQueriesRegistry.java

```diff
@@ -45,6 +46,7 @@ public IndicesQueriesRegistry(Settings settings, Set<QueryParser> injectedQueryP
             }
             namedWriteableRegistry.registerPrototype(queryParser.getBuilderPrototype());
         }
+        namedWriteableRegistry.registerPrototype(EmptyQueryBuilder.PROTOTYPE);
```

Comment:
Will do, I wasn't sure if this was the best place to register it, but I didn't want to add it to the parser registry since we should not allow top-level queries with that name.